source/component/PasDoc_Items.pas

```diff
@@ -1705,6 +1731,88 @@ procedure TPasItem.StoreParamTag(
   ReplaceStr := '';
 end;
 
+function TPasItem.InheritedItem: TPasItem;
+begin
+  if Assigned(MyObject) then
+    Result := MyObject.FindItemInAncestors(Name)
+  else
+    Result := nil;
+end;
+
+function TPasItem.GetInheritedItemDescriptions: TStringPairVector;
+
+  function IsOverrideItem(Item: TPasItem): Boolean;
+  begin
+    Result := (
+     ((Item is TPasRoutine) and (SD_OVERRIDE in TPasRoutine(Self).Directives)) or
```

Comment:
I think above you meant `TPasRoutine(Item)` not `TPasRoutine(Self)`. 

Ah, but I see it doesn't matter -- you always call `IsOverrideItem(Self)`.

Another idea then: make IsOverrideItem a protected virtual method in TPasItem. From what I understand, you can make it functionally equivalent to current code, but avoid "is" checks and typecasts:

- Let it just return false in TPasItem.
- In TPasRoutine let it return `SD_OVERRIDE in Directives`
- In TPasProperty let it return `PropType = ''`